specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2022-10-01/offers.json

```diff
@@ -0,0 +1,302 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHCI/clusters/{clusterName}/publishers/{publisherName}/offers/{offerName}": {
+      "get": {
+        "operationId": "Offers_Get",
+        "description": "Get Offer resource details within a publisher of HCI Cluster.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PublisherNameParameter"
+          },
+          {
+            "$ref": "#/parameters/OfferNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ODataExpandParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Offer"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Offer": {
+            "$ref": "./examples/GetOffer.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "OfferList": {
+      "description": "List of Offer proxy resources for the HCI cluster.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of Offer proxy resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Offer"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to the next set of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Offer": {
+      "description": "Offer details.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Offer properties.",
+          "$ref": "#/definitions/OfferProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "OfferProperties": {
+      "description": "Publisher properties.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
```

Comment:
>provisioningState [](http://example.com/codeflow?start=9&length=17)

Use string enum for this. Must include terminal states (Succeeded, Failed, Canceled). Transition states are up to you.

Reply:
Since these are browse api's and provisioningState object is kind of mandate for all resource types so we added this obj as string type with default value set to 'succeeded'. Do you want us to change to enum and define only one value(which is 'succeeded') ?

Reply:
If your backing resource doesn't communicate any provisioning states, I don't think you should include the property here. Since your APIs don't support PUT/PATCH/DELETE, it's only meaningful if you can translate similar state semantics from your backing resource (e.g., what value do you provide here if someone GETs the resource while it is in the middle of being PUT or PATCHed?). If you decide you want to keep it, you need to define it as string enum with at least the 3 terminal states: Succeeded, Failed, and Canceled. You are welcome to add additional transition states as needed.

Reply:
Actually, this spec has been already part of 
https://github.com/Azure/azure-rest-api-specs-pr/blob/RPSaaSMaster/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-04-01-preview/offers.json#L219

and rolled out for public preview, shall it be acceptable to modify now 
will that not be considered as breaking change?

Reply:
This is why we ask you to include the current/latest API version files as your first commit. Reviews can't tell what to review if you don't do that. This appeared to be a newly added file/API in this PR, so I reviewed it as such.

I'm OK with leaving this as string for this API version, given that it's already published as unrestricted string. In your next API version, I think you should be able to change it to string enum (requires breaking change review) without impacting customers.